File: web/frontend/src/types/patient.ts
export interface PatientSummary {
  id: string;
  name: string;
  dob: string;
  gender: string;
}

export interface Allergy {
  substance: string;
  reaction: string;
  severity: string;
}

export interface Medication {
  name: string;
  dosage: string;
  frequency: string;
  route: string;
  start_date: string | null;
  end_date: string | null;
  prescribing_doctor: string;
  notes: string;
}

export interface Diagnosis {
  icd10_code: string;
  name: string;
  date_diagnosed: string;
  status: string;
  notes: string;
  confidence: number;
}

export interface LabResult {
  test_name: string;
  value: number | string;
  unit: string;
  reference_range: string;
  date: string;
  is_abnormal: boolean;
  notes: string;
}

export interface VitalSigns {
  timestamp: string;
  systolic_bp: number | null;
  diastolic_bp: number | null;
  heart_rate: number | null;
  temperature: number | null;
  spo2: number | null;
  respiratory_rate: number | null;
  weight: number | null;
  height: number | null;
  blood_glucose: number | null;
}

export interface PatientFull {
  id: string;
  first_name: string;
  last_name: string;
  full_name: string;
  age: number;
  date_of_birth: string;
  gender: string;
  blood_type: string | null;
  allergies: Allergy[];
  medications: Medication[];
  diagnoses: Diagnosis[];
  lab_results: LabResult[];
  vitals_history: VitalSigns[];
  family_history: string[];
  surgical_history: string[];
  lifestyle: Record<string, string>;
  notes: string;
}

export interface Consultation {
  id: number;
  patient_id: string | null;
  patient_name: string | null;
  specialty: string;
  complaints: string;
  response: Record<string, unknown>;
  date: string;
}

export interface Skill {
  name: string;
  description: string;
}

export interface Specialization {
  id: string;
  name_ru: string;
  name_en: string;
  description: string;
  skills: Skill[];
}

export interface ConsultationSkill {
  name: string;
  description: string;
  tool: string;
}

export interface ConsultationResult {
  doctor: {
    specialty_id: string;
    name: string;
    qualification: string;
  };
  consultation: {
    complaints: string;
    patient_context: string;
    summary: string;
    available_skills: ConsultationSkill[];
    relevant_icd_prefixes: string[];
    recommended_tests: string[];
  };
  instructions: string;
  disclaimer: string;
}

export interface IcdResult {
  code: string;
  title: string;
  definition?: string;
  uri?: string;
}

export interface PubMedArticle {
  pmid: string;
  title: string;
  authors: string;
  journal: string;
  year: string;
  doi?: string;
  url?: string;
}

export interface DiseaseInfo {
  name: string;
  icd_codes: IcdResult[];
  description: string;
  symptoms: string[];
  risk_factors: string[];
}

File: web/backend/schemas/patient.py
"""Pydantic schemas for patient data."""
from pydantic import BaseModel


class AllergySchema(BaseModel):
    id: int | None = None
    substance: str
    reaction: str = ""
    severity: str = "moderate"


class MedicationSchema(BaseModel):
    id: int | None = None
    name: str
    dosage: str
    frequency: str
    route: str = "oral"
    start_date: str | None = None
    end_date: str | None = None
    prescribing_doctor: str = ""
    notes: str = ""


class DiagnosisSchema(BaseModel):
    id: int | None = None
    icd10_code: str
    name: str
    date_diagnosed: str = ""
    status: str = "active"
    notes: str = ""
    confidence: float = 0.0


class LabResultSchema(BaseModel):
    id: int | None = None
    test_name: str
    value: float | str
    unit: str = ""
    reference_range: str = ""
    date: str = ""
    is_abnormal: bool = False
    notes: str = ""


class VitalSignsSchema(BaseModel):
    id: int | None = None
    timestamp: str = ""
    systolic_bp: int | None = None
    diastolic_bp: int | None = None
    heart_rate: int | None = None
    temperature: float | None = None
    spo2: float | None = None
    respiratory_rate: int | None = None
    weight: float | None = None
    height: float | None = None
    blood_glucose: float | None = None


class PatientSummary(BaseModel):
    id: str
    name: str
    dob: str
    gender: str


class PatientResponse(BaseModel):
    id: str
    first_name: str
    last_name: str
    full_name: str
    age: int
    date_of_birth: str
    gender: str
    blood_type: str | None = None
    allergies: list[AllergySchema]
    medications: list[MedicationSchema]
    diagnoses: list[DiagnosisSchema]
    lab_results: list[LabResultSchema]
    vitals_history: list[VitalSignsSchema]
    family_history: list[str]
    surgical_history: list[str]
    lifestyle: dict[str, str]
    notes: str


class RegisterPatientRequest(BaseModel):
    first_name: str
    last_name: str
    date_of_birth: str
    gender: str
    blood_type: str | None = None
    allergies: list[AllergySchema] = []
    family_history: list[str] = []


class AddVitalsRequest(BaseModel):
    systolic_bp: int | None = None
    diastolic_bp: int | None = None
    heart_rate: int | None = None
    temperature: float | None = None
    spo2: float | None = None
    respiratory_rate: int | None = None
    weight: float | None = None
    height: float | None = None
    blood_glucose: float | None = None


class AddLabResultRequest(BaseModel):
    test_name: str
    value: float | str
    unit: str = ""
    reference_range: str = ""


class AddDiagnosisRequest(BaseModel):
    icd10_code: str
    name: str
    status: str = "active"
    notes: str = ""
    confidence: float = 0.0


class AddMedicationRequest(BaseModel):
    name: str
    dosage: str
    frequency: str
    route: str = "oral"
    notes: str = ""


class AddAllergyRequest(BaseModel):
    substance: str
    reaction: str = ""
    severity: str = "moderate"


class UpdatePatientRequest(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    date_of_birth: str | None = None
    gender: str | None = None
    blood_type: str | None = None
    notes: str | None = None
    family_history: list[str] | None = None
    surgical_history: list[str] | None = None
    lifestyle: dict[str, str] | None = None
